FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.DataClient/Api/FhirApiDataClient.cs

```diff
@@ -55,21 +50,29 @@ public sealed class FhirDataClient : IFhirDataClient
             _httpClient.Timeout = Timeout.InfiniteTimeSpan;
         }
 
-        public async Task<FhirElementBatchData> GetAsync(TaskContext context, CancellationToken cancellationToken = default)
+        public async Task<FhirElementBatchData> GetAsync(
+            string resourceType,
```

Comment:
Shall we wrap the parameters into some `FhirSearchInput` class? Then we don't need to change the interface if we  change/add any search parameters in the future

Reply:
Good idea. I've added a `FhirSearchParameters` class to hold all the search parameters.